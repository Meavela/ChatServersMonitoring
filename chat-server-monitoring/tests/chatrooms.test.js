import React from 'react';
import { render } from '@testing-library/react';
import Chatrooms from '../src/Components/chatrooms';
import renderer from 'react-test-renderer';

// Mock the two components of the page
jest.mock('../src/Components/count-chatrooms');
jest.mock('../src/Components/list-chatrooms');

// Check that the Chatrooms page display the right things
test('Display Chatrooms page', () => {
    const {container} = render(<Chatrooms />)
    expect(container.textContent).toMatch('Chatrooms2On servers :http://127.0.0.1:7419/unittesthttp://127.0.0.1:6592/unitest2ChatroomsServerChatroomhttp://127.0.0.1:7419/unittestBesthttp://127.0.0.1:6592/unitest2Looser')
});

// Check that the Chatrooms page don't change
test('Not change Chatrooms page', () => {
    const component = renderer.create(
        <Chatrooms />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();  
});