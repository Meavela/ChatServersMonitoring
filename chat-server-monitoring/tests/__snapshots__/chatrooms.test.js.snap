// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not change Chatrooms page 1`] = `
<div>
  <div>
    <div
      className="MuiPaper-root MuiCard-root count-card MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root title"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Chatrooms
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-h2"
        >
          2
        </h2>
        <div
          className="servers"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            On servers :
          </p>
          <ul
            className="MuiList-root MuiList-padding"
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root servers-items MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  http://127.0.0.1:7419/unittest
                </span>
              </div>
            </div>
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root servers-items MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                >
                  http://127.0.0.1:6592/unitest2
                </span>
              </div>
            </div>
          </ul>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="MuiPaper-root MuiCard-root list-card MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        className="MuiCardHeader-root title"
      >
        <div
          className="MuiCardHeader-content"
        >
          <span
            className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Chatrooms
          </span>
        </div>
      </div>
      <div
        className="MuiCardContent-root list-card-content"
      >
        <div
          className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <table
            aria-label="simple table"
            className="MuiTable-root"
            role={null}
          >
            <thead
              className="MuiTableHead-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Server
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
                  scope="col"
                >
                  Chatroom
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body"
                  role="cell"
                  scope="row"
                >
                  http://127.0.0.1:7419/unittest
                </th>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  Best
                </td>
              </tr>
              <tr
                className="MuiTableRow-root"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body"
                  role="cell"
                  scope="row"
                >
                  http://127.0.0.1:6592/unitest2
                </th>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                >
                  Looser
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
